### ISSUE TITLE
Strings passed to .itemWithPath are not URL encoded

### ISSUE THREAD

**Issue by etovian** [microsoftgraph/msgraph-sdk-java#259]
### Expected behavior
When a file name/path is passed to IGraphServiceClient.itemWithPath, the file name/path should be properly URL encoded by IGraphServiceClient when the input string is concatenated to the URL generated by IGraphServiceClient.

### Actual behavior
Passing a valid file name/path to .itemWIthPath (e.g., "iText in Action 2nd ed - B. Lowagie (Manning, 2011) WW.pdf") will result in a java.net.URISyntaxException, due to an illegal character (in this case, the spaces) when the request is posted.

### Steps to reproduce the behavior
Invoke the following:
```
client
                .users(oneDriveUserId)
                .drive()
                .root()
                .itemWithPath("iText in Action 2nd ed - B. Lowagie (Manning, 2011) WW.pdf")
                .createUploadSession(new DriveItemUploadableProperties())
                .buildRequest()
                .post();
```

**Comment by ZuperKeurZ** [comment]
just ran into the same issue.

**Comment by ThomasBarnekow** [comment]
Roughly a year later, I also ran into the same issue, using the latest version retrieved from the maven repository. That latest version is 1.7.1 from October 29, 2019, which was published more than 4 months after @etovian opened the issue and more than 6 months ago. 

@bilalgce, could you comment on the status? Will this be fixed anytime soon? What's the workaround until it is fixed?

**Comment by ThomasBarnekow** [comment]
@bilalgce, I just tried to fix the issue by URL-encoding the path in the following way:

```
import java.net.URLEncoder;

String encodedDriveItemPath = URLEncoder.encode(driveItemPath, StandardCharsets.UTF_8);
```

However, that does not work as expected, because the '+' character used to encode a space character will not be decoded to a space character again. What works is this:

```
import org.springframework.web.util.UriUtils;

String encodedDriveItemPath = UriUtils.encode(driveItemPath, StandardCharsets.UTF_8);
```

**Comment by baywet** [comment]
Hi everyone,
Thanks for your feedback and your patience on the matter.
This has been fixed with #449 and released in version 2.1.0 as we replaced the URL builder by the one provided by Okhttp.
No need for pre-encoding anymore, you can pass the path directly in the itemWithPath method and the library will do the encoding for you!
Closing.
